src/mltcontroller.cpp

```diff
@@ -736,94 +736,26 @@ void Controller::next(int currentPosition)
 void Controller::setIn(int in)
 {
     if (m_producer && m_producer->is_valid()) {
-        // Adjust filters.
-        bool changed = false;
-        int n = m_producer->filter_count();
-        for (int i = 0; i < n; i++) {
-            QScopedPointer<Filter> filter(m_producer->filter(i));
-            if (filter && filter->is_valid()) {
-                if (QString(filter->get(kShotcutFilterProperty)).startsWith("fadeIn")) {
-                    if (!filter->get(kShotcutAnimInProperty)) {
-                        // Convert legacy fadeIn filters.
-                        filter->set(kShotcutAnimInProperty, filter->get_length());
-                    }
-                    filter->set_in_and_out(in, filter->get_out());
-                    changed = true;
-                } else if (!filter->get_int("_loader") && filter->get_in() == m_producer->get_in()) {
-                    filter->set_in_and_out(in, filter->get_out());
-                    changed = true;
-                }
-            }
+        int delta = in - m_producer->get_in();
+        if (!delta) {
```

Comment:
Instead of checking only for 0 with an early return consider
```
if (delta > 0) {
    m_producer->set("in", in);
    adjustClipFilters(*m_producer, m_producer->get_in(), m_producer->get_out(), delta, 0);
    Controller::refreshConsumer();
}
```

Reply:
Oh wait. a negative delta is OK right? Nevermind